### PR TITLE
Add SPV_KHR_relaxed_extended_instruction

### PR DESCRIPTION
**Pull request by Keenuts**
Gitlab PR:
https://gitlab.khronos.org/spirv/spirv-extensions/-/merge_requests/250

### CODE REVIEW

File: extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+|======
+7+|[[OpExtInstWithForwardRefs]]*OpExtInstWithForwardRefs* +
+ +
+Executes an instruction in an imported set of non-semantic extended
+instructions.
+
+At least one _<id>_ operand must be a forward reference, otherwise,
+*OpExtInst* must be used.
+
+| 5 + variable | TBD
```
Comment (Naghasan) [inline extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc:110]: Missing the opcode value

File: extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+Modifies the list of locations where a forward reference is allowed to add
+the following cases:
+
+ - OpExtInstWithForwardRefs can contain forward references.
+
+Instructions
+~~~~~~~~~~~~
+
+Add the following new instruction:
```
Comment (Naghasan) [inline extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc:98]: Same as https://github.com/KhronosGroup/SPIRV-Registry/pull/253#discussion_r1618909497, we should specify the section.

File: extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc (hunk 1 above)
Comment (Keenuts) [reply extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc:110]: Is there an actual process to reserve en opcode, or shall I just pick a free number? (404 is unused AFAIK).

Comment (Keenuts) [reply extensions/KHR/SPV_KHR_relaxed_extended_instruction.asciidoc]: Got told about the assigned_numbers file in gitlab, will fix the opcode.

**Comment by Keenuts** [comment]
Rebased & correct opcode added! thanks!

**Review by Naghasan — APPROVED** [review]

**Review by alan-baker — APPROVED** [review]

**Comment by Naghasan** [comment]
Merging following discussions on the 2024-05-29 SPIR WG call

**Pull request merged**